### ISSUE TITLE
ioBroker startup does not work after raspberry restart

### ISSUE THREAD

**Issue by Mike1707** [buanet/ioBroker.docker#174]
Hello! 

After a power failure I needed to restart my router and raspberry pi with ioBroker installed on it.

Unfortunately, I was not able to start ioBroker again. In Docker everything seems to be healthy but the logs show the following

```
------------------------------------------------------------
-----          Step 5 of 5: ioBroker startup           -----
------------------------------------------------------------
 
Starting ioBroker...
 
 
Recived termination signal (SIGTERM).
Shutting down ioBroker... 
``` 

I tried to recreate the container (with portainer) and used the latest image which worked already before the restart. After using `docker exec -it iobroker bash` I tried to find some logs and in the `/log/iobroker.current.log` file I found the following: 

```
2021-04-17 21:08:22.323  - info: host.iobroker iobroker.js-controller version 3.2.16 js-controller starting
2021-04-17 21:08:22.329  - info: host.iobroker Copyright (c) 2014-2021 bluefox, 2014 hobbyquaker
2021-04-17 21:08:22.330  - info: host.iobroker hostname: iobroker, node: v12.22.1
2021-04-17 21:08:22.330  - info: host.iobroker ip addresses: 192.168.117.45 fd00::49e3:736d:24d1:9550 2a02:908:4b13:dc80:7ca:4ab1:de69:46a7 fe80::1273:d624:88b2:f7c1 172.18.0.1 fe80::42:a1ff:fe7a:79ae 169.254.87.183 fe80::70c8:4bff:fe80:8780 169.254.146.92 fe80::54be:17ff:fe9d:7e55
2021-04-17 21:08:22.431  - error: host.iobroker Cannot read system.config: null (OK when migrating or restoring)
2021-04-17 21:08:22.578  - info: host.iobroker connected to Objects and States
2021-04-17 21:08:22.610  - info: host.iobroker added notifications configuration of host
2021-04-17 21:08:22.649  - error: host.iobroker Cannot find view "system" for search "instance"
2021-04-17 21:08:22.650  - error: host.iobroker Could not add notifications config of this host: Could not get notifications setup from instances: Cannot find view "system"
2021-04-17 21:08:22.718  - info: host.iobroker Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
2021-04-17 21:08:22.725  - error: host.iobroker Cannot find view "system" for search "host"
2021-04-17 21:08:22.737  - error: host.iobroker Cannot find view "system" for search "state"
2021-04-17 21:08:22.738  - error: host.iobroker Cannot find view "system" for search "instance"
2021-04-17 21:08:22.739  - error: host.iobroker Could not collect system.host.iobroker states to check for obsolete states: Error: Cannot find view "system"
2021-04-17 21:08:22.740  - error: host.iobroker _design/system missing - call node iobroker.js setup
```

On the host system I didn't change anything and of course I have a volume on the host which is mapped to the container path `/opt/iobroker` on which I didn't change anything either. 

I'm not understanding what happened since I'm using your image for a long while with many raspberry restarts.

I would appreciate any help and can provide more information at any time. 

Thanks in advance!
Mike

**Comment by Mike1707** [comment]
I have a few updates and managed to fix it, but maybe my findings will help you for further development or other people with the same issue.

The solution to fix the issue was taken from `https://213.136.68.177/topic/42816/gel%C3%B6st-iobroker-nach-stromausfall-nicht-mehr-erreichbar/29` but the iobroker forum is currently in maintenance, therefore I used the Google cache with the mentioned link. 

The trick was to override the `objects.json` and `states.json` in `/opt/iobroker/iobroker-data/` with older ones from `/opt/iobroker/iobroker-data/backup-objects/`.

After overriding those files and restarting the docker container I found the following in the logs: 

`hash changed or no sources cached => force download of new sources`

I remembered that something similar has been stated in the logs before I recreated the container, as mentioned in the issue description. Therefore, I thought about the following... 

The raspberry restart happened after a power failure and therefore my router needed to restart as well. I guess that ioBroker tried to download something without internet connection after the router restart/power failure. Maybe that lead to corrupted `states.json` and `objects.json` files.

I'm not sure if that is really the case, but maybe it will help you in the future to keep that case in mind for reliable restarts. 

Nevertheless, thanks for this amazing project!

**Comment by buanet** [comment]
Hi Mike,
my solution would have been to deploy a new container and restore your last backup. In some rare cases it could happen that you get a corrupt file or something after a power failure. So its always a good thing to have an actual backup. :)
Thank you for sharing your solution so others can find it when googling. 

Regards,
André
